### ISSUE TITLE
Instructions for setting up .repo based OP-TEE build?

### ISSUE THREAD

**Issue by dnovick** [OP-TEE/optee_os#1600]
So, I've been working with OP-TEE for a while now, trying to get a few platforms working, including a Zedboard, an Altera SoC, and the RPI3. While the RPI3 build is currently one of the supported builds under .repo, the others aren't. There is of course a port of OP-TEE to the Zynq 7000, but at this point I have yet to get it to work, despite much effort (including talking with the author). Also, I am actually writing the Altera port, as it doesn't exist right now.

That said, I'm considering doing the work to get the Zynq 7000 and Altera platforms under the .repo build setup. This will make it easier for me and for others that might want to use OP-TEE on these platforms. Before starting however, I'd like to understand what the approved process is, etc. BTW, I have already discussed getting the Zynq 7000 build under .repo with the author, and he's totally fine with it.

Can somebody, maybe from Linaro, help me understand 1) if this is even something that you all want and 2) what the process is for doing it.

BTW...Apologies if I should have posted this to the Build issues board. I chose this one simply because it seems to get the most traffic.

Thanks,
Dave

**Comment by jforissier** [comment]
Hi @dnovick,

We're certainly happy with anyone contributing build/manifest files for platforms we support in upstream `optee_os`. To do so, you would create pull requests in https://github.com/OP-TEE/manifest and https://github.com/OP-TEE/build.
Things may get a bit complicated if you need to reference some custom repositories such as Gits on personal GitHub accounts. For maintenance reasons, we'd rather avoid referencing hacked versions of boot loaders, ARM TF, kernel etc. But that can be discussed in the PRs.
@jbech-linaro feel free to comment on this.

**Comment by jbech-linaro** [comment]
Hi  @dnovick  @jforissier , I agree with what Jerome said and regarding personal GitHub accounts, we can move things from there to https://github.com/linaro-swg also, which is something we've done in the past with git's that started out at personal pages. Anyhow, thanks for looking into this and please let us know whether you need our help.

**Comment by dnovick** [comment]
Gents,

Thanks for the response...and sorry for the delay in mine...I was heads down getting my OP-TEE demo done on the RPI3 and have been really busy since. For the moment, I've put further work on the Zed and Altera boards on temporary hold, but hope to get back to them at some point. If/when that happens, I'll get back with you regarding process specifics. I definitely have no intention of referencing personal Git repos. I want whatever I do to align with the "bigger picture" wrt OP-TEE and the platforms it supports.

BTW, my demo has been very well received by numerous audiences. Thanks for all the help you and others gave me along the way.

Dave

**Comment by jbech-linaro** [comment]
I'm closing the ticket, either because it has already been answered or that it is no longer relevant or it could be lack of response from the author. Having that said, feel free to re-open the ticket if you have more to add to the ticket.
